### ISSUE TITLE
Default `imagePullSecrets` value in job when using KubernetesRun causes issues with scheduling/scaling

### ISSUE THREAD

**Issue by gmalkov** [PrefectHQ/prefect#5099]
We were able to resolve this issue, but figured we'd drop an issue here in case anyone else runs into it as it was particularly annoying to track down.

## Description
We are using an EKS managed node group that scales from 0 to run the Prefect job pods. We use `KubernetesRun` and a job template to define the job configuration. The template does not contain any values for `imagePullSecrets`, and we do not pass any overrides for it into the `KubernetesRun` invocation.

When we try to run the flow, we see the job get created and spawn a pod that gets stuck in `Pending`. We see the following error message in the scheduler logs:
```
"Error updating pod" err="failed to create merge patch for pod \"[namespace]\"/\"[pod name]\": map: map[] does not contain declared merge key: name" pod="[namespace]/[pod name]"
```
There is no mention of this pod at all in the cluster autoscaler logs, which is unusual for a pending pod.

We were able to achieve the expected behavior by adding the line `imagePullSecrets: []` to our job template. This resulted in the job created by Prefect not having a value for `imagePullSecrets` at all. Before we made this change, the `imagePullSecrets` value of the job generate by Prefect was:
```
imagePullSecrets:
    - {}
```

## Expected Behavior
The cluster autoscaler should scale up the corresponding managed node group and the pod should get scheduled and executed on a node in that group.

## Reproduction
It would take too long to pare down our configuration to a functional working example for reproduction, especially since we were able to resolve the issue.

## Environment
- EKS, k8s `1.21`
- Prefect `0.15.7`
- Managed node group scaling from 0

**Comment by madkinsz** [comment]
I believe this is being addressed in https://github.com/PrefectHQ/prefect/pull/5001

**Comment by gmalkov** [comment]
> I believe this is being addressed in #5001
Ah, so it is, awesome!
